packages/webcert/src/feature/certificate/UnifiedEdit/UeInteger/UeInteger.tsx

```diff
@@ -0,0 +1,107 @@
+import {
+  CertificateDataElement,
+  ConfigUeInteger,
+  QuestionValidationTexts,
+  TextInput,
+  ValidationError,
+  ValueInteger,
+} from '@frontend/common'
+import { ValidationWrapper } from '@frontend/common/src/components/Inputs/DatePickerCustom/Styles'
+import * as React from 'react'
+import { useEffect, useState } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import styled from 'styled-components'
+import { updateCertificateDataElement, updateClientValidationError } from '../../../../store/certificate/certificateActions'
+import { getVisibleValidationErrors } from '../../../../store/certificate/certificateSelectors'
+
+const INVALID_NUMBER_PERIOD_ERROR = 'Ange ett värde mellan 0 och 100 %'
+
+const Wrapper = styled.div`
+  display: flex;
+  align-items: center;
+`
+const StyledTextInput = styled(TextInput)`
+  width: 45px;
+  padding: 4px 4px;
+  height: 35px;
+  text-align: center;
+  margin-right: 0.625em;
+`
+export interface Props {
+  question: CertificateDataElement
+  disabled: boolean
+}
+
+const UeInteger: React.FC<Props> = ({ question, disabled }) => {
+  const dispatch = useDispatch()
+  const questionValue = question.value as ValueInteger
+  const questionConfig = question.config as ConfigUeInteger
+  const [number, setNumber] = useState<string | null>(questionValue.value?.toString() ?? '')
+  const validationErrors = useSelector(getVisibleValidationErrors(question.id, 'NUMBER'))
+
+  useEffect(() => {
+    if (questionValue.value) {
+      isNumberValid(questionValue.value)
+    }
+  }, [questionValue.value])
+
+  const onKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === ' ') {
+      event.preventDefault()
+    }
+  }
+
+  const handleNumberOnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const inputValue = event.target.value
+    const numberValue = Number(inputValue)
+
+    if (isNaN(numberValue)) return null
```

Comment:
This prevents you from entering negative numbers by typing.
It also prevents you from deleting a negative number you have pasted in using backspace, unless you select the whole number and delete it all at once.
Bothe cases since '-' isNaN